sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -120,6 +121,19 @@ sealed trait FrameBoundary {
   def notFollows(other: FrameBoundary): Boolean
 }
 
+/**
+ * Extractor for making working with frame boundaries easier.
+ */
+object FrameBoundary {
+  def apply(boundary: FrameBoundary): Option[Int] = unapply(boundary)
+  def unapply(boundary: FrameBoundary): Option[Int] = boundary match {
+    case CurrentRow => Some(0)
+    case ValuePreceding(offset) => Some(-offset)
+    case ValueFollowing(offset) => Some(offset)
+    case _ => None
```

Comment:
How to tell the difference between `UnboundedPreceding` and `UnboundedFollowing`? 

Or this assume that we can't use both of them in same position (start or end)?